tests/openwisp2/routing.py

```diff
@@ -0,0 +1,16 @@
+from channels.auth import AuthMiddlewareStack
+from channels.routing import ProtocolTypeRouter, URLRouter
+from channels.security.websocket import AllowedHostsOriginValidator
+from openwisp_notifications.websockets.routing import (
+    get_routes as get_notification_routes,
+)
+
+from openwisp_controller.geo.channels.routing import geo_routes
+
+application = ProtocolTypeRouter(
+    {
+        'websocket': AllowedHostsOriginValidator(
+            AuthMiddlewareStack(URLRouter(get_notification_routes() + geo_routes))
+        )
+    }
+)
```

Comment:
I think we should also update [openwisp_controller/geo/channels/routing.py]( https://github.com/openwisp/openwisp-controller/blob/master/openwisp_controller/geo/channels/routing.py) and remove openwisp_notifications routes from the file.